fix: replace rgba() with rgb() in code blocks

OnkarRuikar opened the pull request:
Now the CSS `rgba()` function is an alias for the `rgb()` function value. We should stop using `rgba()` henceforth.

OnkarRuikar committed:
fix: replace rgba() with rgb() in code blocks

estelle commented on files/en-us/learn/css/building_blocks/values_and_units/index.md:
```diff
@@ -327,7 +327,7 @@ In the example below, we have added a background image to the containing block o
 
 > [!NOTE]
-> In older versions of CSS, the `rgb()` syntax didn't support an alpha parameter - you needed to use a different function called `rgba()` for that. These days you can pass an alpha parameter to `rgb()`. The `rgba()` function is an alias for `rgb()`.
+> In older versions of CSS, the `rgb()` syntax didn't support an alpha parameter - you needed to use a different function called `rgba()` for that. These days you can pass an alpha parameter to `rgb()`. Now, the `rgba()` function is an alias for `rgb()`.
```
let's get rid of this note. It's beyond what learning of values and units need to know or care about (used to be important when rgb support for a was new)

OnkarRuikar replied:
Yes, enough time has passed that we can stop promoting the old syntax, `rgba()`, and `hsla()`.

estelle reviewed (approved):
i think we should remove the rgba note from learn, but this is an improvement, so approving. thank you.

Josh-Cena reviewed (changes requested):
Why is everything about `rgba` removed, even in the references? If it's part of the web it needs to be documented, even if it's legacy/deprecated. This includes both the old comma syntax and `rgba`.

OnkarRuikar committed:
remove old syntax and rgba and hsla notes

OnkarRuikar committed:
add alias notes

OnkarRuikar commented:
> Why is everything about rgba removed, even in the references? If it's part of the web it needs to be documented, even if it's legacy/deprecated. This includes both the old comma syntax and rgba.

We are removing history parts. We shouldn't promote rgba(), hsla(), and comma syntax as alternatives to the latest stuff. Also, assertions need to be made about avoiding their uses.

I've reverted the change in syntax blocks and added notes at the top of rgb() and hsl() pages.

Josh-Cena commented:
We shouldn't show legacy methods as if you should use them, but we should show that they exist and how they may look—this helps developers identify them, understand their limitations, and migrate off them. I don't know why the examples that clearly say they are legacy and the mere notes that "there's a legacy grammar using comma-separated lists" don't deserve a place on MDN when they are very much a part of both the web and the standards. They are even in the formal syntax.

Josh-Cena commented on files/en-us/web/css/color_value/hsl/index.md:
```diff
@@ -10,6 +10,9 @@ spec-urls:
 {{CSSRef}}
 
+> [!NOTE]
+> The `hsla()` functional notation is an alias for `hsl()`. They both work the same. It is recommend to use `hsl()`.
```
```suggestion
> The `hsla()` functional notation is an alias for `hsl()`. They are exactly equivalent. It is recommend to use `hsl()`.
```

OnkarRuikar replied:
```suggestion
> The `hsla()` functional notation is an alias for `hsl()`. They are exactly equivalent. It is recommended to use `hsl()`.
```

Josh-Cena commented on files/en-us/web/css/color_value/hsl/index.md:
```diff
@@ -31,10 +34,6 @@ hsl(from #0000FF h s calc(l + 20))
 
-The `hsla()` function can also be used to express sRGB colors. This is an alias for `hsl()` that accepts the same parameters.
-
-> **Note:** `hsl()`/`hsla()` can also be written in a legacy form in which all values are separated with commas, for example `hsl(120, 75%, 25%)` or `hsla(120deg, 75%, 25%, 0.8)`. The `none` value is not permitted in the comma-separated legacy syntax, the `deg` on the hue value is optional, and the `%` units are required for the saturation and lightness values.
```
This needs to be either kept or be documented in the syntax box.

OnkarRuikar replied:
done

Josh-Cena commented on files/en-us/web/css/color_value/hsl/index.md:
```diff
@@ -301,63 +297,6 @@ div.comma-separated {
 
 {{EmbedLiveSample("legacy_syntax_comma-separated_values", "100%", 150)}}
 
-### Legacy versus modern syntax
```
This needs to be kept, otherwise no one really knows what the legacy syntax looks like or what exactly it could do.

OnkarRuikar replied:
I've added `hsla()` and legacy format to the to the syntax section. There is no need to show different formats that produce the same output in an example. :thinking:

OnkarRuikar replied:
done. The example is back.

Josh-Cena commented on files/en-us/web/css/css_colors/relative_colors/index.md:
```diff
@@ -171,9 +171,6 @@ rgb(from red r g b / alpha)
 It is worth mentioning again that the color system of the origin color doesn't need to match the color system being used to create the output color. Again, this provides a lot of flexibility. Generally you won't be interested in and might not even know the system the origin color is defined in (you might just have a [custom property value](#using_custom_properties) to manipulate). You'll just wan ...
 
-> [!NOTE]
-> Aliases such as `rgba()` or `hsla()` can be used to output relative colors, and to specify origin colors. When using legacy color functions to output a relative color, you must use the comma-less modern syntax and can't mix percentages and numbers.
```
So I've seen this note several times but I'm not sure what it's saying. Does that mean `hsl` and `rgb` support relative colors even with the comma-separated syntax? Or in fact `hsl`, `hsla`, `rgb`, `rgba` all require comma-less syntax in order to use relative colors? It looks like it's the latter, in which case it should be documented as "a reason you should use comma-less".

OnkarRuikar replied:
Relative color syntax doesn't accept commas in any case. The note is just being a distraction. We need to simply remove it.

Josh-Cena reviewed (commented):
Here are everything I feel like should be explained more

OnkarRuikar commented on files/en-us/web/css/color_value/rgb/index.md:
```diff
@@ -10,6 +10,9 @@ spec-urls:
 {{CSSRef}}
 
+> [!NOTE]
+> The `rgba()` functional notation is an alias for `rgb()`. They both work the same. It is recommend to use `rgb()`.
```
```suggestion
> The `rgba()` functional notation is an alias for `rgb()`. They are exactly equivalent. It is recommended to use `rgb()`.
```

OnkarRuikar committed:
Apply suggestions from code review

OnkarRuikar committed:
cover old stuff in syntax section

Josh-Cena commented on files/en-us/web/css/color_value/rgb/index.md:
````diff
@@ -25,11 +28,14 @@ rgb(255 255 255 / 50%)
-```
 
-The `rgba()` function can also be used to express sRGB colors. This is an alias for `rgb()` that accepts the same parameters.
+* Legacy 'rgba()' alias */
````
```suggestion
/* Legacy 'rgba()' alias */
```

Josh-Cena commented on files/en-us/web/css/color_value/hsl/index.md:
```diff
@@ -10,6 +10,9 @@ spec-urls:
 {{CSSRef}}
 
+> [!NOTE]
+> The `hsla()` functional notation is an alias for `hsl()`. They are exactly equivalent. It is recommended to use `hsl()`.
```
I think this can be omitted—it doesn't have to be _this_ visible. (Same for `rgba` obviously)

OnkarRuikar replied:
The `hsla()` page url redirects to this page so the people looking for hsla() information need to know right away what happened to it.

Josh-Cena commented on files/en-us/web/css/color_value/hsl/index.md:
```diff
@@ -29,9 +32,14 @@ hsl(none 75% 25%)
+
+/* Legacy format */
+hsl(120, 75%, 25%)
+hsla(120deg, 75%, 25%, 0.8)
```
```suggestion
hsl(120deg, 75%, 25%, 0.8)
```

Since we already established that hsla is an alias, let's just use `hsl`

OnkarRuikar replied:
We've covered this case already on line 28 and extra hsla sample is no biggie. :)

Josh-Cena replied:
Line 28 is the comma-less syntax. I'm saying here we should consistently demonstrate comma-list syntax using `hsl` because `hsla` is just exactly equivalent, and it's better to be consistent where variation is not the point.

Josh-Cena commented on files/en-us/web/css/color_value/rgb/index.md:
```diff
@@ -25,11 +28,14 @@ rgb(255 255 255 / 50%)
-> **Note:** `rgb()`/`rgba()` can also be written in a legacy form in which all values are separated with commas, for example `rgb(255, 0, 0)`. Mixing number and percent value types is not valid in the comma-separated legacy syntax (i.e. the `R`, `G`, and `B` values must be either all numbers or all percentages), and the `none` value is also not permitted.
+/* Legacy format */
+rgb(0, 255, 255)
+rgba(0, 255, 255, 50%)
```
```suggestion
rgb(0, 255, 255, 50%)
```

OnkarRuikar replied:
`rgb(0, 255, 255, 50%)` case has already been covered on line 25

Josh-Cena replied:
Line 25 is the comma-less syntax though?

Josh-Cena commented on files/en-us/web/css/css_values_and_units/index.md:
```diff
@@ -318,7 +318,7 @@ The {{cssxref("&lt;position&gt;")}} type defines 2D positioning of an object ins
 White space is allowed, but optional inside the parentheses. (But see notes regarding whitespace within pages for `min()`, `max()`, `minmax()`, and `clamp()` functions.)
 
-Some legacy functional notations, such as legacy syntax for `rgb()`, `rgba()`, `hsl()`, and `hsla()`, used commas, but commas are generally only used to separate items in a list. If a comma is used to separate arguments, white space is optional before and after the comma.
+Commas are generally only used to separate items in a list. If a comma is used to separate arguments, white space is optional before and after the comma.
```
Mention that colors don't need commas, because not everyone will know this

OnkarRuikar replied:
The section is not about colors, they were used only as an example. Using `linear-gradient()` as an example now:

```suggestion
Some functional notations, such as [`linear-gradient()`](/en-US/docs/Web/CSS/gradient/linear-gradient) use commas, but commas are generally only used to separate items in a list. If a comma is used to separate arguments, white space is optional before and after the comma.
```

Josh-Cena reviewed (approved):
Generally LGTM, with a few comments to consider

OnkarRuikar committed:
Apply suggestions from code review

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

OnkarRuikar committed:
Apply suggestions from code review

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

Josh-Cena commented:
@estelle could you take another look and merge if it's good?

OnkarRuikar commented:
ping

bsmth commented on files/en-us/web/css/css_values_and_units/index.md:
```diff
@@ -318,7 +318,7 @@ The {{cssxref("&lt;position&gt;")}} type defines 2D positioning of an object ins
 White space is allowed, but optional inside the parentheses. (But see notes regarding whitespace within pages for `min()`, `max()`, `minmax()`, and `clamp()` functions.)
 
-Some legacy functional notations, such as legacy syntax for `rgb()`, `rgba()`, `hsl()`, and `hsla()`, used commas, but commas are generally only used to separate items in a list. If a comma is used to separate arguments, white space is optional before and after the comma.
+Some functional notations, such as [`linear-gradient()`](/en-US/docs/Web/CSS/gradient/linear-gradient) use commas, but commas are generally only used to separate items in a list. If a comma is used to separate arguments, white space is optional before and after the comma.
```
I think there's no harm in the note about legacy syntax here, what do you reckon?

OnkarRuikar replied:
The note is not about the legacy syntax. It is about the optional white spaces when commas are used in functional notation values. The `rgb()` and `rgba()` functions are used just as an example. So there is no harm in using a new example and future proofing the content by removing `rgb()` and `rgba()`.

OnkarRuikar replied:
I don't agree, but reverting to get the PR merged.
```suggestion
Some legacy functional notations, such as legacy syntax for `rgb()`, `rgba()`, `hsl()`, and `hsla()`, used commas, but commas are generally only used to separate items in a list. If a comma is used to separate arguments, white space is optional before and after the comma.
```

OnkarRuikar committed:
put back the old vs new syntax example

OnkarRuikar committed:
fix more occurances since the PR was created

OnkarRuikar committed:
Update files/en-us/web/css/css_values_and_units/index.md

bsmth reviewed (approved):
Thank you! I think we should merge to unblock the work 👍🏻
